app/controllers/application_controller.rb

```diff
@@ -12,7 +12,12 @@ def include_google_analytics?
   end
 
   def set_visitor_id
-    if cookies[:visitor_id].present?
+    return if current_intake&.visitor_id.present? && current_intake&.visitor_id == cookies[:visitor_id]
+
+    if current_intake&.visitor_id.present?
```

Comment:
also realized that we were checking cookies and not overwriting visitor_id on intakes but not checking intakes first - does this seem right? happy to talk it over in a call because it's a little confusing imo

Reply:
okay, I wrote out a big thing here but I'm not going to post it because I realized I don't have context.

I suspect the logic here can be simplified if we set the `visitor_id` when the Intake is created, in `Questions::FeelingsController`. This would be nice because it's also the same as how we set `source` and `referrer`.

I think this logic would be simpler, I think it could be accomplished like this:

```ruby
def set_visitor_id
  if current_intake&.visitor_id.present?
    cookies.permanent[:visitor_id] = current_intake.visitor_id
  else
    cookies.permanent[:visitor_id] ||= SecureRandom.hex(26)
  end
end
```

But yeah, I am very likely missing some context or behavior that needs to happen here. Happy to chat.